google/cloud/pubsub/internal/tracing_pull_ack_handler.cc

```diff
@@ -30,11 +32,43 @@ namespace pubsub_internal {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
 #ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
-
 using Attributes =
     std::vector<std::pair<opentelemetry::nostd::string_view,
                           opentelemetry::common::AttributeValue>>;
+using Links =
+    std::vector<std::pair<opentelemetry::trace::SpanContext, Attributes>>;
+
+namespace {
+Links CreateLinks() {
+#if OPENTELEMETRY_ABI_VERSION_NO >= 2
+  auto consumer_span_context =
+      opentelemetry::trace::GetSpan(
+          opentelemetry::context::RuntimeContext::GetCurrent())
+          ->GetContext();
+  if (consumer_span_context.IsSampled() && consumer_span_context.IsValid()) {
+    return Links{{consumer_span_context, Attributes{}}};
+  }
+#endif
+  return Links{};
+}
+
+void MaybeAddLinkAttributes(
+    opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> const& span) {
+#if OPENTELEMETRY_ABI_VERSION_NO < 2
+  auto consumer_span_context =
+      opentelemetry::trace::GetSpan(
+          opentelemetry::context::RuntimeContext::GetCurrent())
+          ->GetContext();
+  if (consumer_span_context.IsSampled() && consumer_span_context.IsValid()) {
+    span->SetAttribute("gcp_pubsub.receive.trace_id",
+                       internal::ToString(consumer_span_context.trace_id()));
+    span->SetAttribute("gcp_pubsub.receive.span_id",
+                       internal::ToString(consumer_span_context.span_id()));
+  }
+#endif
```

Comment:
Avoid "unused argument" warnings:

```suggestion
  }
#else
  (void)span;
#endif
```